### ISSUE TITLE
Failed email delivery crashes the reporter, which Tower re-reports — creating an infinite loop

### ISSUE THREAD

**Issue by jamauro** [mimiquate/tower_email#96]
When email delivery fails, the reporter crashes — and because Tower captures that crash, it tries to email the crash report too. If that delivery also fails, each failure spawns the next attempt, indefinitely. We hit this in production: one failed report grew into a sustained loop of ~10 crashing send tasks per second until we manually ran `Tower.detach()`.

Two things combine to cause it:

**1. A failed delivery crashes the send task**

`TowerEmail.Reporter.send_email/4` asserts success:

```elixir
async(fn ->
  {:ok, _} = TowerEmail.Mailer.deliver(email_message)
end)
```

Any `{:error, _}` from the adapter (rate limit, rejected message, network problem) raises a `MatchError` in the task. Tower captures the crash and reports it — through the same email path that just failed.

**2. Multiline subjects make delivery failure permanent**

The subject is built from the exception message, truncated but not sanitized. Many exception messages span several lines (`MatchError` always does), and some providers reject newlines in subjects outright — Resend returns:

```
422 The `\n` is not allowed in the `subject` field.
```

So the crash report from problem 1 can never be delivered, which keeps the loop going: every attempt fails with 422, crashes, and gets re-reported.

### Suggested fixes

- In `send_email/4`, handle `{:error, reason}` by logging (at a level Tower won't capture) instead of crashing. A reporter whose failure feeds back into the pipeline it reports through shouldn't be able to crash.
- In `TowerEmail.Message`, collapse whitespace in the subject to a single line before truncating.

Both are small; happy to send a PR if useful.

Seen on tower 0.8.8 / tower_email 0.6.0 with the Resend adapter, and both spots look unchanged on `main`.

**Comment by grzuy** [comment]
Hi @jamauro,

Thank you for taking the time to report this issue!

Indeed, we need a fix.

This is analogous to what was reported in `tower_rollbar` here: https://github.com/mimiquate/tower_rollbar/issues/115.

> happy to send a PR if useful

That would be great.
If we could come up with a similar improvement to the one in tower_rollbar.

Thank you 🙏
